runner/src/runner.rs

```diff
@@ -626,4 +687,55 @@ mod tests {
         expected_output.push('\n');
         assert_eq!(output.stdout, expected_output.as_bytes());
     }
+
+    fn set_mocks(
+        runner: &mut MockRunner,
+        shutdown_flag: Arc<AtomicBool>,
+        shutdown_flag_count: usize,
+        terminate_proc_count: usize,
+    ) {
+        let tmp = TempDir::new("runner-tests").expect("failed to create tempdir");
+        let mock_path = tmp.path().to_path_buf();
+        runner.expect_subscribe_to_shutdown_signals().returning(|| Ok(()));
+        runner.expect_download_path().return_const(mock_path.clone());
+        runner
+            .expect_try_get_release()
+            .returning(|_| Ok(Some(ClientRelease::default())));
+        runner.expect_download_binary().returning(|_| Ok(()));
+        runner.expect_run_binary().returning(|| Ok(()));
+        runner
+            .expect_shutdown_flag()
+            .times(shutdown_flag_count)
+            .return_const(shutdown_flag);
+        // expect this to be called once
+        runner
+            .expect_terminate_proc_and_wait()
+            .times(terminate_proc_count)
+            .returning(|| Ok(()));
+
+        let downloaded_release = DownloadedRelease {
+            release: ClientRelease {
+                uri: String::default(),
+                code_hash: H256::default(),
+            },
+            path: mock_path.clone(),
+            bin_name: String::default(),
+        };
+        runner
+            .expect_downloaded_release()
+            .return_const(Some(downloaded_release));
+        runner.expect_delete_downloaded_release().returning(|| Ok(()));
+    }
+
+    #[tokio::test]
+    #[should_panic(expected = "MockRunner::shutdown_flag: Expectation(<anything>) called more than 2 times")]
+    async fn test_runner_terminate_child_procs_on_shutdown() {
+        let mut runner = MockRunner::default();
+        set_mocks(&mut runner, Arc::new(AtomicBool::new(true)), 1, 1);
```

Comment:
There are already starting to be quite a few parameters here. I think it's still fine but let me know if I should use the builder pattern here.

Reply:
Why not have `set_mocks` simply return the `MockRunner`?